### PR TITLE
Use arrow functions in the service worker

### PR DESCRIPTION
**Pull request by marco-c**

### CODE REVIEW

**Comment by marco-c** [comment]
Ah, tests are failing on Node.js v0.12 because it doesn't support arrow functions.

File: templates/app/offline-worker.js [hunk 1]
```diff
@@ -54,52 +56,44 @@
...
+          }
+          console.error('Error fetching ' + url + ', status was ' + response.status);
+        });
+      })));
     },
 
     // Remove the offline caches not controlled by this worker.
     clearOtherCaches: function () {
-      var deleteIfNotCurrent = function (cacheName) {
+      return self.caches.keys()
```
Comment (delapuente) [inline templates/app/offline-worker.js:84]: I would prefer to keep the name as I think it adds readability. We can transform it into an arrow function but keeping the name:

``` js
var deleteIfNotCurrent = cacheName => {
  if (cacheName.indexOf(this.CACHE_PREFIX !== 0 || cacheName === this.CACHE_NAME) {
    return Promise.resolve();
  }
  return self.caches.delete(cacheName);
};

return self.caches.keys()
.then(keys => Promise.all(keys.map(deleteIfNotCurrent)));
```

Comment (marco-c) [inline templates/app/offline-worker.js:84]: I originally did this, then made it an anonymous function and added a comment.
I can add it back if you prefer.

File: test/service-worker/testCacheResources.js [hunk 2]
```diff
@@ -29,14 +27,17 @@ describe('cacheResources()', function () {
 
     self.oghliner.RESOURCES = [];
     sinon.stub(Date, 'now').returns(testTimestamp);
-    sinon.stub(self, 'fetch').returns(Promise.resolve(mockedOkResponse));
-    sinon.stub(oghliner, 'prepareCache').returns(Promise.resolve(mockCache));
+    sinon.stub(self, 'fetch', function(request) {
```
Comment (delapuente) [inline test/service-worker/testCacheResources.js]: These changes are not related to this PR. Do you mind to drop them? We can discuss them in another PR.

Comment (marco-c) [inline test/service-worker/testCacheResources.js]: Removed it.

File: templates/app/offline-worker.js (hunk 1 above)
Comment (delapuente) [inline templates/app/offline-worker.js:84]: I missed the comment. Maybe we can write:

``` js
.then(cacheNames => Promise.all(cacheNames.map(cacheName => {
  var isAPreviousCache = cacheName.startsWith(this.CACHE_PREFIX) && cacheName !== this.CACHE_NAME;
  return isAPreviousCache ? self.caches.delete(cacheName) : Promise.resolve();
});
```

What do you think (I tend to delay the use of comments as much as possible). You can use `.startsWith()`. It is not ES5 but, you know, this PR is about converting into arrow functions. ;)

**Comment by delapuente** [comment]
Cool, let's merge.

**Comment by delapuente** [comment]
@marco-c the [build is failing](https://travis-ci.org/mozilla/oghliner/builds/103666609) but there are no differences between the HEAD of the repository and 47ce4b3 which is ok. I'm testing locally and I cannot reproduce the error with latest Node. Can you take a look?

**Comment by marco-c** [comment]
@delapuente it was an intermittent issue.

**Comment by delapuente** [comment]
Ok, it failed me three times in a row but it's good to hear and makes sense. ;)
